chips/stm32f429zi/src/ethernet/mod.rs

```diff
@@ -0,0 +1,2328 @@
+
+    fn test_mac_default_values(ethernet: &Ethernet) {
+        assert_eq!(EthernetSpeed::Speed100Mbs, ethernet.get_speed());
+        assert_eq!(false, ethernet.is_loopback_mode_enabled());
+        assert_eq!(OperationMode::FullDuplex, ethernet.get_operation_mode());
+        assert_eq!(false, ethernet.is_mac_transmiter_enabled());
+        assert_eq!(false, ethernet.is_mac_receiver_enabled());
+        assert_eq!(false, ethernet.is_address_filter_enabled());
+        assert_eq!(false, ethernet.is_mac_tx_full());
+        assert_eq!(true, ethernet.is_mac_tx_empty());
+        assert_eq!(false, ethernet.is_mac_tx_writer_active());
+        assert_eq!(MacTxReaderStatus::Idle, ethernet.get_mac_tx_reader_status());
+        assert_eq!(false, ethernet.is_mac_tx_in_pause());
+        assert_eq!(MacTxWriterStatus::Idle, ethernet.get_mac_tx_writer_status());
+
+        assert_eq!(RxFifoLevel::Empty, ethernet.get_rx_fifo_fill_level());
+        assert_eq!(MacRxReaderStatus::Idle, ethernet.get_mac_rx_reader_status());
+        assert_eq!(false, ethernet.is_mac_rx_writer_active());
+
+        assert_eq!(false, ethernet.is_mac_mii_active());
+        assert_eq!(DEFAULT_MAC_ADDRESS, ethernet.get_mac_address0());
+        assert_eq!(false, ethernet.is_mac_address1_enabled());
+        assert_eq!(false, ethernet.is_mac_address2_enabled());
+        assert_eq!(false, ethernet.is_mac_address3_enabled());
+    }
+
+    fn test_dma_default_values(ethernet: &Ethernet) {
+        assert_eq!(
+            &ethernet.transmit_descriptor as *const TransmitDescriptor as u32,
+            ethernet.get_transmit_descriptor_list_address()
+        );
+        assert_eq!(
+            DmaTransmitProcessState::Stopped,
+            ethernet.get_transmit_process_state()
+        );
+        assert_eq!(true, ethernet.is_transmit_store_and_forward_enabled());
+        assert_eq!(false, ethernet.is_dma_transmission_enabled());
+        assert_eq!(
+            DmaTransmitThreshold::Threshold64,
+            ethernet.get_dma_transmission_threshold_control()
+        );
+
+        assert_eq!(
+            &ethernet.receive_descriptor as *const ReceiveDescriptor as u32,
+            ethernet.get_receive_descriptor_list_address()
+        );
+        assert_eq!(true, ethernet.is_receive_store_and_forward_enabled());
+        assert_eq!(false, ethernet.is_dma_reception_enabled());
+        assert_eq!(
+            DmaReceiveThreshold::Threshold64,
+            ethernet.get_dma_receive_threshold_control()
+        );
+
+        assert_eq!(false, ethernet.did_normal_interrupt_occur());
+        assert_eq!(false, ethernet.did_abnormal_interrupt_occur());
+    }
+
+    /// Test Ethernet initialization
+    pub fn test_ethernet_init(ethernet: &Ethernet) {
+        debug!("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
+        debug!("Testing Ethernet initialization...");
+
+        // This is broken for some reasons
+        assert_eq!(Ok(()), ethernet.init());
```

Comment:
If this is broken, we probably shouldn't ship this test? Do you know what the underlying root cause is?